In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("./../Datasets/zomato.csv")

In [3]:
df.head()

Unnamed: 0.1  Unnamed: 0     restaurant name restaurant type  \
0             0           0        #FeelTheROLL     Quick Bites   
1             1           1          #L-81 Cafe     Quick Bites   
2             2           2             #refuel            Cafe   
3             3           3  '@ Biryani Central   Casual Dining   
4             4           4          '@ The Bbq   Casual Dining   

   rate (out of 5)  num of ratings  avg cost (two people) online_order  \
0              3.4               7                  200.0           No   
1              3.9              48                  400.0          Yes   
2              3.7              37                  400.0          Yes   
3              2.7             135                  550.0          Yes   
4              2.8              40                  700.0          Yes   

  table booking                                      cuisines type  \
0            No                                          Fast Food   
1            No                               Fast Food, Beverages   
2            No                                    Cafe, Beverages   
3            No                          Biryani, Mughlai, Chinese   
4            No  BBQ, Continental, North Indian, Chinese, Bever...   

                             area      local address  
0                       Bellandur          Bellandur  
1  Byresandra,Tavarekere,Madiwala                HSR  
2               Bannerghatta Road  Bannerghatta Road  
3                    Marathahalli       Marathahalli  
4                       Bellandur          Bellandur

In [4]:
df.columns

Index(['Unnamed: 0.1', 'Unnamed: 0', 'restaurant name', 'restaurant type',
       'rate (out of 5)', 'num of ratings', 'avg cost (two people)',
       'online_order', 'table booking', 'cuisines type', 'area',
       'local address'],
      dtype='object')

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7105 entries, 0 to 7104
Data columns (total 12 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Unnamed: 0.1           7105 non-null   int64  
 1   Unnamed: 0             7105 non-null   int64  
 2   restaurant name        7105 non-null   object 
 3   restaurant type        7105 non-null   object 
 4   rate (out of 5)        7037 non-null   float64
 5   num of ratings         7105 non-null   int64  
 6   avg cost (two people)  7048 non-null   float64
 7   online_order           7105 non-null   object 
 8   table booking          7105 non-null   object 
 9   cuisines type          7105 non-null   object 
 10  area                   7105 non-null   object 
 11  local address          7105 non-null   object 
dtypes: float64(2), int64(3), object(7)
memory usage: 666.2+ KB


In [6]:
df.dtypes

Unnamed: 0.1               int64
Unnamed: 0                 int64
restaurant name           object
restaurant type           object
rate (out of 5)          float64
num of ratings             int64
avg cost (two people)    float64
online_order              object
table booking             object
cuisines type             object
area                      object
local address             object
dtype: object

In [7]:
df.tail()

Unnamed: 0.1  Unnamed: 0         restaurant name     restaurant type  \
7100          7100        7100                  Zoey's                Cafe   
7101          7101        7101  ZOROY Luxury Chocolate      Dessert Parlor   
7102          7102        7102       Zu's Doner Kebaps  Takeaway, Delivery   
7103          7103        7103                   Zyara       Casual Dining   
7104          7104        7104                  Zyksha          Food Truck   

      rate (out of 5)  num of ratings  avg cost (two people) online_order  \
7100              4.3             894                  600.0          Yes   
7101              4.0              68                  250.0          Yes   
7102              3.7              33                  350.0           No   
7103              3.8             191                  650.0          Yes   
7104              3.4               9                  200.0           No   

     table booking                         cuisines type               area  \
7100            No    Cafe, Italian, Continental, Burger          Bellandur   
7101            No                              Desserts       Brigade Road   
7102            No  Turkish, Fast Food, Biryani, Chinese       Malleshwaram   
7103            No        North Indian, Mughlai, Chinese       Kammanahalli   
7104            No                             Fast Food  Bannerghatta Road   

        local address  
7100    Sarjapur Road  
7101    Church Street  
7102         RT Nagar  
7103       HBR Layout  
7104  South Bangalore

In [8]:
columns_to_remove = ['Unnamed: 0.1', 'Unnamed: 0']  # List of columns to remove
df.drop(columns=columns_to_remove,inplace=True)


In [9]:
df.head()

restaurant name restaurant type  rate (out of 5)  num of ratings  \
0        #FeelTheROLL     Quick Bites              3.4               7   
1          #L-81 Cafe     Quick Bites              3.9              48   
2             #refuel            Cafe              3.7              37   
3  '@ Biryani Central   Casual Dining              2.7             135   
4          '@ The Bbq   Casual Dining              2.8              40   

   avg cost (two people) online_order table booking  \
0                  200.0           No            No   
1                  400.0          Yes            No   
2                  400.0          Yes            No   
3                  550.0          Yes            No   
4                  700.0          Yes            No   

                                       cuisines type  \
0                                          Fast Food   
1                               Fast Food, Beverages   
2                                    Cafe, Beverages   
3                          Biryani, Mughlai, Chinese   
4  BBQ, Continental, North Indian, Chinese, Bever...   

                             area      local address  
0                       Bellandur          Bellandur  
1  Byresandra,Tavarekere,Madiwala                HSR  
2               Bannerghatta Road  Bannerghatta Road  
3                    Marathahalli       Marathahalli  
4                       Bellandur          Bellandur

In [10]:
df.shape

(7105, 10)

In [11]:
df.isnull().sum()

restaurant name           0
restaurant type           0
rate (out of 5)          68
num of ratings            0
avg cost (two people)    57
online_order              0
table booking             0
cuisines type             0
area                      0
local address             0
dtype: int64

In [12]:
df.dropna().shape

(6984, 10)

In [13]:
df.dropna(inplace=True)

In [13]:
df.shape

(7105, 10)

In [14]:
restaurant_type_value_counts = df['restaurant type'].value_counts()

In [16]:
restaurant_type_value_counts

Quick Bites              2840
Casual Dining            1634
Cafe                      403
Delivery                  358
Takeaway, Delivery        289
                         ... 
Microbrewery, Lounge        1
Bakery, Beverage Shop       1
Bar, Pub                    1
Quick Bites, Mess           1
Club, Casual Dining         1
Name: restaurant type, Length: 81, dtype: int64

In [17]:
#keeping in dataframe to view properly
restaurant_type_value_counts = restaurant_type_value_counts.reset_index()
restaurant_type_value_counts.columns = ['restaurant type', 'count']
restaurant_type_value_counts


restaurant type  count
0             Quick Bites   2840
1           Casual Dining   1634
2                    Cafe    403
3                Delivery    358
4      Takeaway, Delivery    289
..                    ...    ...
76   Microbrewery, Lounge      1
77  Bakery, Beverage Shop      1
78               Bar, Pub      1
79      Quick Bites, Mess      1
80    Club, Casual Dining      1

[81 rows x 2 columns]

### There are 82 unique types of restaurant

### lets check top 20 restaurants that have most orders


In [20]:
restaurant_type_value_counts.head(20)

restaurant type  count
0                  Quick Bites   2840
1                Casual Dining   1634
2                         Cafe    403
3                     Delivery    358
4           Takeaway, Delivery    289
5               Dessert Parlor    217
6                       Bakery    154
7           Casual Dining, Bar    123
8                Beverage Shop    118
9                          Bar     82
10                  Food Court     80
11                 Fine Dining     79
12                      Lounge     58
13          Bar, Casual Dining     54
14                        Mess     48
15                  Sweet Shop     46
16                       Kiosk     36
17                         Pub     34
18     Sweet Shop, Quick Bites     31
19  Beverage Shop, Quick Bites     29

### Let us check bottom 10 restaurant type

In [21]:
restaurant_type_value_counts.tail(20)

restaurant type  count
61      Quick Bites, Meat Shop      1
62   Fine Dining, Microbrewery      1
63         Beverage Shop, Cafe      1
64   Food Court, Beverage Shop      1
65     Quick Bites, Food Court      1
66           Microbrewery, Bar      1
67   Casual Dining, Sweet Shop      1
68                  Bhojanalya      1
69        Lounge, Microbrewery      1
70  Dessert Parlor, Food Court      1
71                Lounge, Cafe      1
72  Food Court, Dessert Parlor      1
73  Casual Dining, Quick Bites      1
74                Cafe, Lounge      1
75                   Cafe, Bar      1
76        Microbrewery, Lounge      1
77       Bakery, Beverage Shop      1
78                    Bar, Pub      1
79           Quick Bites, Mess      1
80         Club, Casual Dining      1

---
 <span style="color:red">*From above we get to know that types that are mixed variety or are having raw supplies are less ordered from*</span>
---
 



Analyzing the information provided by the `df.info()` output is a crucial step in understanding the structure and content of your DataFrame. Here's a breakdown of the information and some steps you can take to analyze it:

1. **Column Information:**
   - **restaurant name:** Categorical (object) column with 6984 non-null values.
   - **restaurant type:** Categorical (object) column with 6984 non-null values.
   - **rate (out of 5):** Numeric (float64) column with 6984 non-null values.
   - **num of ratings:** Numeric (int64) column with 6984 non-null values.
   - **avg cost (two people):** Numeric (float64) column with 6984 non-null values.
   - **online_order:** Categorical (object) column with 6984 non-null values.
   - **table booking:** Categorical (object) column with 6984 non-null values.
   - **cuisines type:** Categorical (object) column with 6984 non-null values.
   - **area:** Categorical (object) column with 6984 non-null values.
   - **local address:** Categorical (object) column with 6984 non-null values.

2. **Analysis Steps:**
   Here are some steps you can take to analyze this information:

   - **Data Types:** Verify if the data types for each column are appropriate. For example, the columns seem to have correct data types (object for text, int64 for integer, float64 for floating-point numbers).

   - **Missing Values:** You've mentioned that there are no missing values (non-null counts match the total count of rows, which is 6984). However, in a real-world scenario, it's a good practice to check for missing values and handle them appropriately.

   - **Categorical Columns:** For categorical columns like 'restaurant type', 'online_order', 'table booking', 'cuisines type', 'area', and 'local address', you could examine unique values using `df['column_name'].unique()` and count the occurrences of each category using `df['column_name'].value_counts()`.

   - **Numeric Columns:** For numeric columns like 'rate (out of 5)', 'num of ratings', and 'avg cost (two people)', you can use basic summary statistics like mean, median, minimum, maximum, and standard deviation. You can calculate these using functions like `df['column_name'].mean()`, `df['column_name'].median()`, `df['column_name'].min()`, `df['column_name'].max()`, and `df['column_name'].std()`.

   - **Correlation:** You can explore the correlation between numeric columns using `df.corr()` to see how different numeric variables are related to each other.

   - **Data Distribution:** Plot histograms or box plots to visualize the distribution of numeric columns. For categorical columns, you can create bar plots to visualize the distribution of categories.

   - **Insights:** Based on the analysis, you can derive insights about the dataset. For example, you might find which types of restaurants are more popular, whether there's a correlation between ratings and the number of ratings, or how the average cost varies across different areas.

Remember, the specific analyses you perform depend on the goals and questions you have about the dataset. Each step aims to give you a better understanding of the data and its patterns.

In [30]:
#Getting to know what categorical values are having values
categorical_columns = ['restaurant type', 'online_order', 'table booking', 'cuisines type', 'area', 'local address']
subset = df.loc[0:2, categorical_columns]
subset

restaurant type online_order table booking         cuisines type  \
0     Quick Bites           No            No             Fast Food   
1     Quick Bites          Yes            No  Fast Food, Beverages   
2            Cafe          Yes            No       Cafe, Beverages   

                             area      local address  
0                       Bellandur          Bellandur  
1  Byresandra,Tavarekere,Madiwala                HSR  
2               Bannerghatta Road  Bannerghatta Road

### Getting all unique values


In [36]:
len(df['area'].unique())

30

#### This means there are 30 regions from where food has been ordered

In [40]:
#Cuisine type
numOfCuisines = len(df['cuisines type'].unique())
print("There are ",numOfCuisines,"different kind of food cuisine")
print("Which are most popular food type:")

df['cuisines type'].value_counts()

There are  2175 different kind of food cuisine
Which are most popular food type:


North Indian, Chinese                            421
North Indian                                     420
South Indian                                     348
Fast Food                                        140
South Indian, North Indian, Chinese              128
                                                ... 
Seafood, Andhra                                    1
Beverages, Cafe, Desserts, Spanish, Bakery         1
Goan, Seafood, North Indian, Chinese, Biryani      1
Healthy Food, Cafe, Beverages                      1
Turkish, Fast Food, Biryani, Chinese               1
Name: cuisines type, Length: 2175, dtype: int64

#### So north indian is most popular followed by chinese and South Indian, whereas seafoods comes among the least favourite

In [43]:
#Lets see more in detail which food type is least or more popular
cuisine=df['cuisines type'].value_counts()

In [44]:
#This gives list of most favourite food types
cuisine.head(20)

North Indian, Chinese                  421
North Indian                           420
South Indian                           348
Fast Food                              140
South Indian, North Indian, Chinese    128
Biryani                                122
Bakery, Desserts                        97
Desserts                                87
Cafe                                    85
Bakery                                  77
North Indian, South Indian, Chinese     73
Chinese, North Indian                   70
Chinese                                 69
North Indian, South Indian              63
North Indian, Chinese, Biryani          57
South Indian, North Indian              56
Mithai, Street Food                     54
Street Food                             48
Finger Food                             45
Kerala                                  42
Name: cuisines type, dtype: int64

In [45]:
#This gives list of least food types
cuisine.tail(20)

Italian, Chinese, Continental, Steak                          1
Chinese, Hyderabadi, Mughlai, South Indian                    1
Italian, Salad, Healthy Food, Cafe, Asian, Desserts, Pizza    1
Seafood, South Indian, Chinese, Kerala                        1
Biryani, Hyderabadi, Andhra, North Indian                     1
Cafe, Burger, Continental, Italian, Desserts                  1
Cafe, Continental, Italian, Beverages, Mediterranean          1
South Indian, Bakery, Chinese                                 1
Italian, Pizza, Fast Food, Salad                              1
South Indian, Biryani, Mangalorean                            1
Continental, Pizza, Burger                                    1
North Indian, Mughlai, Seafood, Chinese                       1
Arabian, Chinese, North Indian, Kerala, Seafood               1
Mithai, Street Food, Fast Food                                1
South Indian, Kerala, Seafood                                 1
Seafood, Andhra                         

In [48]:
#lets check for online orders stats
df['online_order'].value_counts()

Yes    3727
No     3378
Name: online_order, dtype: int64

#### So we can say that online and offline orders are nearly the same

### Handling numereic data

In [50]:
df.describe()

rate (out of 5)  num of ratings  avg cost (two people)
count      7037.000000     7105.000000            7048.000000
mean          3.514253      188.921042             540.286464
std           0.463249      592.171049             462.902305
min           1.800000        1.000000              40.000000
25%           3.200000       16.000000             300.000000
50%           3.500000       40.000000             400.000000
75%           3.800000      128.000000             600.000000
max           4.900000    16345.000000            6000.000000

In [65]:
# Group by restaurant name and calculate mean rating and number of ratings
ratings_info = df.groupby('restaurant name')['rate (out of 5)'].agg(['mean', 'count'])

# Rename the columns for clarity
ratings_info.rename(columns={'mean': 'mean-rating', 'count': 'number of rating'}, inplace=True)

# Print the first few rows of the ratings information
ratings_info.head(30)


mean-rating  number of rating
restaurant name                                                         
#FeelTheROLL                                       3.4                 1
#L-81 Cafe                                         3.9                 1
#refuel                                            3.7                 1
'@ Biryani Central                                 2.7                 1
'@ The Bbq                                         2.8                 1
'@99                                               3.4                 1
'@Italy                                            4.1                 1
'@North Parontha Hut                               2.8                 1
1000 B.C                                           3.2                 1
100°C               3.7                 1
11 to 11 Express Biriyanis                         3.5                 1
1131 Bar + Kitchen                                 4.4                 1
12th Main - Grand Mercure                          4.1                 1
1441 Pizzeria                                      4.1                 1
1522 - The Pub                                     4.2                 1
154 Breakfast Club                                 4.0                 1
1722 Urban Bistro                                  4.1                 1
18+ Ice Cafe                                       3.5                 1
1947                                               3.9                 1
1980s Games Cafe                                   3.4                 1
1992 Chats - Space                                 3.7                 1
1Q1                                                4.3                 1
1TO3 Kitchen                                       3.1                 1
2 Statez                                           3.7                 1
20 Char - Sterlings MAC Hotel                      4.0                 1
20 Feet High                                       4.0                 1
21 C - La Marvella Sarovar Premiere Hotel          3.6                 1
24 Carats - The Capitol                            3.1                 1
24 Hours Cake Delivery                             3.3                 1
24 Hours Coffee Drop - La Classic                  3.4                 1

### This means that we have all unique restaurant name/restaurant in dataset

In [66]:
 ratings_info.sort_values(by='mean-rating', ascending=False,inplace=True)
 ratings_info.head(10)


mean-rating  \
restaurant name                                                   
SantÃÃÃÂ...          4.9   
Asia Kitchen By Mainland China                              4.9   
Byg Brewski Brewing Company                                 4.9   
The Pizza Bakery                                            4.8   
Punjab Grill                                                4.8   
O.G. Variar & Sons                                          4.8   
Belgian Waffle Factory                                      4.8   
Biergarten                                                  4.7   
Taaza Thindi                                                4.7   
The Black Pearl                                             4.7   

                                                    number of rating  
restaurant name                                                       
SantÃÃÃÂ...                 1  
Asia Kitchen By Mainland China                                     1  
Byg Brewski Brewing Company                                        1  
The Pizza Bakery                                                   1  
Punjab Grill                                                       1  
O.G. Variar & Sons                                                 1  
Belgian Waffle Factory                                             1  
Biergarten                                                         1  
Taaza Thindi                                                       1  
The Black Pearl                                                    1

In [62]:
len(df['restaurant name'].unique())

7105

In [68]:
df.columns

Index(['restaurant name', 'restaurant type', 'rate (out of 5)',
       'num of ratings', 'avg cost (two people)', 'online_order',
       'table booking', 'cuisines type', 'area', 'local address'],
      dtype='object')

In [82]:
cost_restaurant = df.sort_values(by='avg cost (two people)', ascending=False)

In [92]:
columns=["restaurant name","avg cost (two people)"]
cost_restaurant.loc[0:10,columns,inplce=True]

SyntaxError: invalid syntax (2694807074.py, line 2)